.github/workflows/rust.yml

```diff
@@ -0,0 +1,42 @@
+name: Rust
+
+on:
+  push:
+  pull_request:
+    branches: [ "main" ]
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  format:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Format code
+      run: cargo fmt
```

Comment:
`cargo fmt` formats the files - and formatted files will just be discarded after this job finishes. We need something that either:
- checks the formatting and fails the CI if it's wrong. You can achieve that with `cargo fmt --check`
- formats the files and commits it. I don't like this option (it's more complicated to set up, it creates additional commits making history ugly), so let's stick with `--check`.

Reply:
scylla-rust-driver and cpp-rust-driver use `cargo fmt --verbose --all -- --check` - let's to the same.
@piodul do you know why there is another double-dash before `--check`? It seems to work the same without it. If there is no difference, let's use `cargo fmt --verbose --all --check`

Reply:
```
piodul@scyllapc ~]$ cargo fmt --help
cargo-fmt
This utility formats all bin and lib files of the current crate using rustfmt.

USAGE:
    cargo fmt [OPTIONS] [-- <rustfmt_options>...]

ARGS:
    <rustfmt_options>...    Options passed to rustfmt
```

It looks like `cargo fmt` calls `rustfmt` underneath. As I understand it, `cargo fmt` might not understand all flags that `rustfmt` accepts so the double-dash can be used to just forward the flags to `rustfmt`.

It indeed looks like `--check` is supported by `cargo fmt`:

```
        --check
            Run rustfmt in check mode
```

so yeah, let's drop the double dash. Maybe `--check` wasn't supported in the past?